### ISSUE TITLE
Node preset - automatic server restart not working? 

### ISSUE THREAD

**Issue by Maxim-Filimonov** [mozilla-neutrino/neutrino-dev#950]
### Bug or issue?

Please try to answer the following questions:

- What version of Neutrino are you using?
8.2.3
- Are you trying to use any presets? If so, which ones, and what versions?
Node - 8.2.3
- Are you using the Yarn client or the npm client? What version?
npm - 10
- What version of Node.js are you using?
node - 10
- What operating system are you using?
Mac Os El Capitan
- What did you do?
1. Created a new project using create-project.
2. Added express to the project and wire it up in app.js file.
3. start server with npm start
4. add new route(see [this line](https://github.com/Maxim-Filimonov/bike-route-server/blob/server_restart_issue/src/app.js#L7)
Got
```
✔ Build completed
[HMR] Updated modules:
[HMR]  - ./src/app.js
[HMR] Update applied.
```
5. Try to access the route - it did not work.
- What did you expect to happen?
To be able to access the new route
- What actually happened, contrary to your expectations?
I am not able to access the new route until I stop npm start and restart it.

**Comment by constgen** [comment]
HMR replacement of Express or Koa application requires a special approach. These both frameworks are very similar. I can show you the Gist how to do this in Koa2: https://gist.github.com/constgen/f1f640c8421befa596c2f58ad3d839d4

For Express you just need to change a line where an app instance is created
**from**
```js
currentApp = require('./app').callback();
```
**to**
```js
currentApp = require('./app');
```
I beleave

It looks different in Express because it doesn't have a `.callback()` method and the application itself is designed to be a request handler.

Try it!

**Comment by Maxim-Filimonov** [comment]
Looks like there is something similar for express - https://github.com/darul75/express-hot-reload

**Comment by constgen** [comment]
Did you change the code for Express in multiple lines? Because there are several spots in **index.js** with
```js
currentApp = require('./app').callback();
```

**Comment by Maxim-Filimonov** [comment]
Yeah see [here](https://github.com/Maxim-Filimonov/bike-route-server/commit/0521c05e26cb3f0de4617cb505438787c09d1cfa)

**Comment by constgen** [comment]
You used EcmaScript modules spec in one of files. Are you sure that this works in your NodeJS? I suggest you to use CommonJS.

**Comment by edmorley** [comment]
The `@neutrinojs/node` preset configures `yarn start` out of the box, which uses [start-server-webpack-plugin](https://github.com/ericclemmons/start-server-webpack-plugin). Is there something that's not working using that, or is it that you'd prefer more prod-dev parity and so want to use express for development too? (Just trying to work out which of question/docs bug/start server bug/feature request this might be)

**Comment by edmorley** [comment]
@Maxim-Filimonov don't suppose you saw the above?

**Comment by Maxim-Filimonov** [comment]
Hi @edmorley. Yeah, missed it. 
We would prefer to run the same server in prod/dev, is it not possible ?

**Comment by edmorley** [comment]
I didn't say it wasn't possible, just trying to figure out the reason for using it :-)

It sounds like this issue is more a question/possible docs addition. I don't personally use Express/Koa, but someone else might have the answer.
